fhirr4/ballerina/src/main/resources/fhirservice/http_service_builder.bal

```diff
@@ -58,6 +74,23 @@ isolated function getHttpService(Holder h, r4:ResourceAPIConfig apiConfig) retur
                         return processSearch;
                     }
                     r4:FHIRContext fhirContext = check r4:getFHIRContext(ctx);
+                    if authzConfig is r4:AuthzConfig {
+                        // handle smart security
+                        string resourceType = split[split.length() - 1];
+                        string? patientID = ();
+                        if resourceType == PATIENT_RESOURCE {
+                            patientID = req.getQueryParamValue(PATIENT_ID_QUERY_PARAM);
+                        } else {
+                            patientID = req.getQueryParamValue(PATIENT_QUERY_PARAM);
+                        }
+                        r4:FHIRSecurity? fhirSecurity = fhirContext.getFHIRSecurity();
+                        if fhirSecurity is r4:FHIRSecurity {
+                            r4:FHIRError? handleSmartSecurityResult = r4:handleSmartSecurity(patientID, authzConfig.authzServiceUrl, fhirSecurity, authzConfig.privilegedClaimUrl);
```

Comment:
Shall we pass authzConfig. that way later if something to be added/modified in authzconfig we can change the handleSmartSecurity implementation accordingly

Reply:
done